.github/pull_request_template.md

```diff
@@ -1,23 +1,79 @@
+-->
+- [ ] My code follows the [style guidelines of this project](CONTRIBUTING.md#java-language-formatting-guidelines)
+- [ ] I have rebased my changes on top of the latest master branch
+<!--
+Update your local repository with the most recent code from the main repo, and rebase your branch on top of the latest master branch. We prefer your initial changes to be squashed into a single commit. Later, if we ask you to make changes, add them as separate commits. This makes them easier to review. As a final step before merging we will either ask you to squash all commits yourself or we'll do it for you.
+-->
+- [ ] I have performed a self-review of my own code
+- [ ] I have made corresponding changes to the documentation
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/master/CONTRIBUTING.md#testing) pass locally with my changes
+<!--
+Run the test suite to make sure that nothing is broken. See https://github.com/elastic/apm-agent-java/blob/master/CONTRIBUTING.md#testing for details.
+-->
+- [ ] I have updated [CHANGELOG.asciidoc](CHANGELOG.asciidoc)
+- [ ] I have updated [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)
+- [ ] Added an API method or config option? Document in which version this will be introduced.
+- [ ] Added an instrumentation plugin? How did you make sure that old, non-supported versions are not instrumented by accident?
 
-If this pull request is work in progress, create a draft PR instead of prefixing the title with WIP.
+## Author's Checklist
+<!-- _(Recommended)_
+Add a checklist of things that are required to be reviewed in order to have the PR approved
+-->
+
+## How to test this PR locally
+<!-- _(Recommended)_
+Explain here how this PR will be tested by the reviewer: commands, dependencies, steps, etc.
+-->
 
+## Related issues
+<!-- _(Recommended)_
+Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it. For more info see:
+https://help.github.com/articles/closing-issues-using-keywords/
+- Closes #ISSUE_ID
+- Relates #ISSUE_ID
+- Requires #ISSUE_ID
+- Superseeds #ISSUE_ID
 -->
 
-### Checklist
+## Use cases
+<!-- _(Recommended)_
+Explain here the different behaviors that this PR introduces or modifies in this project, user roles, environment configuration, etc.
+If you are familiar with Gherkin test scenarios, we recommend its usage: https://cucumber.io/docs/gherkin/reference/
+-->
 
-<!-- Potential tasks related to a new PR. Remove tasks that are not relevant -->
+## Logs
```

Comment:
Could you explain what the purpose of this is? 🙂

Reply:
It could be a stacktrace coming from an application server, or a Java framework/library, database logs... bringing more information about the potential issue to be fixed... which brings to my mind that it could be helpful for bugs and maybe not that useful for new features. wdyt?

What's more important than the structure is your feeling of usefulness of the template, so please feel free to adapt/delete any section in it

Reply:
Ah, got it. I think I'd remove this from the PR template then as it feels a better place for it is the issue for the bug report.

Reply:
Do you mind if I move it to the issue template then?

Reply:
Never mind, it's already there, thanks!